src/components/FederationInviteShower.tsx

```diff
@@ -0,0 +1,55 @@
+import { Copy, QrCode } from "lucide-solid";
+import { createSignal, Show } from "solid-js";
+import { QRCodeSVG } from "solid-qr-code";
+
+import { SimpleDialog } from "~/components";
+import { useI18n } from "~/i18n/context";
+import { useCopy } from "~/utils";
+
+export function FederationInviteShower(props: {
+    name?: string;
+    inviteCodes: string[];
+}) {
+    const i18n = useI18n();
+    const [showQr, setShowQr] = createSignal(false);
+
+    const [copy, copied] = useCopy({ copiedTimeout: 1000 });
+
+    return (
+        <>
+            <div class="flex w-full justify-center gap-8">
+                <button onClick={() => setShowQr(true)}>
+                    <QrCode class="inline-block h-4 w-4" />
+                </button>
+                <button
+                    class="p-1"
+                    classList={{
+                        "bg-m-red rounded": copied()
+                    }}
+                    onClick={() => copy(props.inviteCodes[0])}
```

Comment:
Would be cool if you could scroll through the invite codes, in case the first isn't working

Reply:
we don't actually have multiple. made it singular